python/ray/llm/_internal/serve/deployments/prefill_decode_disagg/prefill_decode_disagg.py

```diff
@@ -0,0 +1,198 @@
+"""Using Ray Serve to deploy LLM models with P/D disaggregation.
+"""
+import asyncio
+import logging
+from typing import AsyncGenerator, Union
+import uuid
+
+from pydantic import BaseModel
+from vllm.config import KVTransferConfig
+
+from ray import serve
+from ray.llm._internal.serve.configs.prompt_formats import Prompt
+from ray.llm._internal.serve.configs.server_models import (
+    LLMRawResponse,
+    parse_args as parse_llm_configs,
+)
+from ray.llm._internal.serve.deployments.llm.llm_server import ResponsePostprocessor
+from ray.llm._internal.serve.deployments.llm.vllm.vllm_models import (
+    KV_TRANSFER_PARAMS_KEY,
+)
+from ray.serve.deployment import Application
+from ray.serve.handle import DeploymentHandle
+from ray.serve.llm import (
+    LLMConfig,
+    LLMRouter,
+    LLMServer,
+    ModelLoadingConfig,
+    build_llm_deployment,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class PDServingArgs(BaseModel):
+    """Schema for P/D serving args."""
+
+    prefill_config: Union[str, LLMConfig]
+    decode_config: Union[str, LLMConfig]
+
+    def parse_args(self) -> "PDServingArgs":
+        """Converts this LLMServingArgs object into an DeployArgs object."""
+
+        def parse_configs_and_cast_type(config: Union[str, LLMConfig]) -> LLMConfig:
```

Comment:
please add a TODO here to clean this up. We need to refactor `parse_llm_configs` to make it more modular and reusable in different contexts.